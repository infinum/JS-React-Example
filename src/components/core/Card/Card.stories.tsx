import React, { ComponentProps } from 'react';
import { StoryFn, Meta } from '@storybook/react';

import { Card } from './Card';

export default {
	title: 'UI/core/Card',
	component: Card,
} as Meta;

const Template: StoryFn<ComponentProps<typeof Card>> = (args) => <Card {...args} />;

export const Basic = Template.bind({});
Basic.args = {
	/* the args you need here will depend on your component */
};
