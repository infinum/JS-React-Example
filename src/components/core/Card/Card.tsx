import { chakra, forwardRef, StyleProps } from '@chakra-ui/react';

export type ICardProps = StyleProps;

export const Card = forwardRef<ICardProps, 'div'>((props, ref) => {
	return (
		<chakra.div
			bg="chakra-body-bg"
			borderRadius={{ base: 'md', sm: 'xl' }}
			_dark={{
				shadow: { base: 'md', sm: 'md-dark' },
			}}
			_light={{ shadow: 'md' }}
			ref={ref}
			{...props}
		/>
	);
});
