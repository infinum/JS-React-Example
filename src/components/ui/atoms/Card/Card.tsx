import { chakra, forwardRef, omitThemingProps, ThemingProps, useStyleConfig, HTMLChakraProps } from '@chakra-ui/react';
import { cx, __DEV__ } from '@chakra-ui/utils';
import React from 'react';

export interface ICardProps extends HTMLChakraProps<'div'>, ThemingProps {}

export const Card = forwardRef<ICardProps, 'div'>(function Card(props, ref) {
	const styles = useStyleConfig('Card', props);

	const { className, ...rest } = omitThemingProps(props);

	const _className = cx('card', className);

	return <chakra.div ref={ref} className={_className} {...rest} __css={styles} />;
});

if (__DEV__) {
	Card.displayName = 'Card';
}
