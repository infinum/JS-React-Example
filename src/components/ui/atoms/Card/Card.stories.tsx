import React, { ComponentProps } from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { Card } from './Card';

export default {
	title: 'UI/Atoms/Card',
	component: Card,
} as Meta;

const Template: Story<ComponentProps<typeof Card>> = (args) => <Card {...args} />;

export const Example = Template.bind({});
Example.args = {
	/* the args you need here will depend on your component */
	children:
		'Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.',
};
