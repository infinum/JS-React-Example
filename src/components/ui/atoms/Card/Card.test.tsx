import React from 'react';

import { Card } from './Card';
import { render } from '@test-utils';

describe('Card', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Card />);

    expect(asFragment()).toMatchSnapshot();
  });
});