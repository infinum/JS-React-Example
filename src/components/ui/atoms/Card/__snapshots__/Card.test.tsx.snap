// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  border: 1px solid;
  border-color: #E2E8F0;
  border-radius: 0.375rem;
  padding: 0.75rem;
}

<div
    class="card emotion-0"
  />
</DocumentFragment>
`;
